src/test/scala/outwatch/ScenarioTestSpec.scala

```diff
@@ -193,38 +200,61 @@ class ScenarioTestSpec extends JSDomSpec {
 
     list.childElementCount shouldBe 0
 
-    val todo = "fold laundry"
-    inputElement.value = todo
-    inputElement.dispatchEvent(inputEvt)
-    submitButton.dispatchEvent(clickEvt)
-
-    list.childElementCount shouldBe 1
-
-    val todo2 = "wash dishes"
-    inputElement.value = todo2
-    inputElement.dispatchEvent(inputEvt)
-    submitButton.dispatchEvent(clickEvt)
-
-    list.childElementCount shouldBe 2
-
-    val todo3 = "clean windows"
-    inputElement.value = todo3
-    inputElement.dispatchEvent(inputEvt)
-    submitButton.dispatchEvent(clickEvt)
-
-    list.childElementCount shouldBe 3
+    (for {
 
-    document.getElementById(todo2).dispatchEvent(clickEvt)
-
-    list.childElementCount shouldBe 2
+      todo <- Task {
+        val todo = "fold laundry"
+        inputElement.value = todo
+        inputElement.dispatchEvent(inputEvt)
+        submitButton.dispatchEvent(clickEvt)
+        todo
+      }
+      _ <- Task {
+        list.childElementCount shouldBe 1
+      }.executeWithFork
```

Comment:
This is because Monix uses by default a [batched execution model](https://monix.io/docs/2x/execution/scheduler.html#execution-model), so we need to "fork" the assertion to the end of the javascript execution queue, otherwise it might execute before all the effects of the event dispatch execute.